Foxy update documentation (#43)
ros2-realtime-demo/pendulum: README.md

## Disclaimer

This is work in progress and should be considered just as a proof of concept on how to leverage ROS
 2 features to use in real-time applications. Any feedback and help to improve the project is
  appreciated.

**Foxy branch is under active development. The current status can be checked here: 
https://github.com/ros2-realtime-demo/pendulum/issues/35**

## Build status

> ![Build Status](https://github.com/ros2-realtime-demo/pendulum/workflows/build/badge.svg?branch=foxy) [![License](https://img.shields.io/badge/license-Apache%202-blue)]() ![GitHub issues](https://img.shields.io/github/issues/ros2-realtime-demo/pendulum)

Reviewer: Why this change?

Author: I thought that makes more sense to only show the build status for the current branch. For every release, we will be adding a new badge, also if we stop supporting an old branch and it stops passing I would prefer not to show it. I don't have a strong opinion on this, change this as you think is best.

Reviewer: Fair enough.

  work made by the Open Robotics. The project uses these packages as a base.

This is the motivation of using an inverted pendulum for a real-time demo:

>A classic example of a controls problem commonly solved by real-time computing is balancing an inverted pendulum. If the controller blocked for an unexpectedly long amount of time, the pendulum would fall down or go unstable. But if the controller reliably updates at a rate faster than the motor controlling the pendulum can operate, the pendulum will successfully adapt react to sensor data to balance the pendulum.

A more detailed description of demo the project design be found here: [Design Article](docs/design.md)

## Install instructions

 Several ways to install the pendulum demo project are provided depending on the user needs. 
 
- [ADE based installation](docs/installation.md#ade#based#installation): This provides a 
pre-configured Docker image with everything ready to be used. This is the recommended option to
 quickly test the demo. It is also useful to have development environment to add new features.
- [Installation from source](docs/installation.md#installation#from#source): provides instructions

>  to build all the required ROS 2 packages from source

Reviewer: ```suggestion
 to build all the required ROS 2 packages from source.
```

- [Crosscompilation](docs/installation.md#crosscompile): provides instructions to

>  crosscompile the demo for an embedded device

Reviewer: ```suggestion
 crosscompile the demo for an embedded device.
```

- [Pre-configured Raspberry PI image](docs/installation.md#raspberry#pi#image): provides a

>  Raspberry PI with everything configured in a Linux Real-time

Reviewer: ```suggestion
 Raspberry PI with everything configured in a Linux Real-time (RT-PREEMPT patch).
```

## How to run the demo

### Source your workspace

If installed as an ADE image:

```bash

> source /opt/ros/foxy/setup.bash

Reviewer: This will change with the upcoming work.

Author: Right, if we merge this first, the instructions should be consistent with the current status. The instructions should be changed accordingly with the upcoming work.

Reviewer: I'll include a file on the volume that source it.

```

If installed from source:

```bash
cd ~/pendulum_ws
source ./install/setup.bash
```

### Launch the demo

Launch the demo with rviz enabled:

```bash
ros2 launch pendulum_bringup pendulum_bringup.launch.py rviz:=True
```

If everything went well you should see the inverted pendulum being controlled in rviz:

![pendulum_rviz](docs/images/pendulum_rviz.gif)

## How to configure real-time settings<a name="rtdemo"></a>

Instructions to configure the demo real-time settings can be found [here](docs/real_time_tutorial.md).

## How to measure real-time performance metrics<a name="rtmetrics"></a>

TODO

## How to contribute

Make a fork using github, and make your changes. This project uses the same guidelines as all the

>  officials ROS2 packages, make sure that your code is compliant with cppcheck, uncrustify...

Reviewer: ```suggestion
 officials ROS 2 packages, make sure that your code is compliant with cppcheck, uncrustify...
```
